drivers/led/is31fl3194.c

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT issi_is31fl3194
+
+/**
+ * @file
+ * @brief IS31FL3194 LED driver
+ *
+ * The IS31FL3194 is a 3-channel LED driver that communicates over I2C.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/dt-bindings/led/led.h>
+
+LOG_MODULE_REGISTER(is31fl3194, CONFIG_LED_LOG_LEVEL);
+
+#define IS31FL3194_PROD_ID_REG	0x00
+#define IS31FL3194_CONF_REG	0x01
+#define IS31FL3194_OUT_EN_REG	0x02
+#define IS31FL3194_CURRENT_REG	0x03
+#define IS31FL3194_OUT1_REG	0x10
+#define IS31FL3194_OUT2_REG	0x21
+#define IS31FL3194_OUT3_REG	0x32
+#define IS31FL3194_UPDATE_REG	0x40
+
+#define IS31FL3194_PROD_ID_VAL		0xce
+#define IS31FL3194_CONF_ENABLE		0x01
+#define IS31FL3194_CONF_RGB_EN		0x04
+#define IS31FL3194_OUT_EN_OUT1		0x01
+#define IS31FL3194_OUT_EN_OUT2		0x02
+#define IS31FL3194_OUT_EN_OUT3		0x04
+#define IS31FL3194_UPDATE_VAL		0xc5
+
+#define IS31FL3194_CHANNEL_COUNT 3
+
+static const uint8_t led_channels[] = {
+	IS31FL3194_OUT1_REG,
+	IS31FL3194_OUT2_REG,
+	IS31FL3194_OUT3_REG
+};
+
+struct is31fl3194_config {
+	struct i2c_dt_spec bus;
+	uint8_t num_leds;
+	const struct led_info *led_infos;
+	const uint8_t *current_limits;
+};
+
+static const struct led_info *is31fl3194_led_to_info(const struct is31fl3194_config *config,
+						     uint32_t led)
+{
+	if (led < config->num_leds) {
+		return &config->led_infos[led];
+	}
+
+	return NULL;
+}
+
+static int is31fl3194_get_info(const struct device *dev,
+			       uint32_t led,
+			       const struct led_info **info_out)
+{
+	const struct is31fl3194_config *config = dev->config;
+	const struct led_info *info = is31fl3194_led_to_info(config, led);
+
+	if (!info) {
+		return -EINVAL;
+	}
+
+	*info_out = info;
+	return 0;
+}
+
+static int is31fl3194_set_color(const struct device *dev, uint32_t led, uint8_t num_colors,
+			     const uint8_t *color)
+{
+	const struct is31fl3194_config *config = dev->config;
+	const struct led_info *info = is31fl3194_led_to_info(config, led);
+	int ret;
+
+	if (!info) {
+		return -ENODEV;
+	}
+
+	if (info->num_colors != 3) {
+		return -ENOTSUP;
+	}
+	if (num_colors != 3) {
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < 3; i++) {
+		uint8_t value;
+
+		switch (info->color_mapping[i]) {
+		case LED_COLOR_ID_RED:
+			value = color[0];
+			break;
+		case LED_COLOR_ID_GREEN:
+			value = color[1];
+			break;
+		case LED_COLOR_ID_BLUE:
+			value = color[2];
+			break;
+		default:
+			/* unreachable: mapping already tested in is31fl3194_check_config */
+			continue;
+		}
+
+		ret = i2c_reg_write_byte_dt(&config->bus, led_channels[i], value);
+		if (ret) {
+			break;
+		}
+	}
+	if (!ret) {
```

Comment:
ret is not a bool,  see: https://pvs-studio.com/en/docs/warnings/v2584/ and fix everything according to that.